bchain/coins/lux/luxrpc.go

```diff
@@ -0,0 +1,190 @@
+        Method string `json:"method"`
+}
+type ResGetBlockChainInfo struct {
+        Error  *bchain.RPCError `json:"error"`
+        Result struct {
+                Chain         string      `json:"chain"`
+                Blocks        int         `json:"blocks"`
+                Headers       int         `json:"headers"`
+                Bestblockhash string      `json:"bestblockhash"`
+                Difficulty struct {
+                       ProofOfWork  json.Number `json:"proof-of-work"`
+                       ProofOfStake json.Number `json:"proof-of-stake"`
+                }
+                SizeOnDisk    int64       `json:"size_on_disk"`
+                Warnings      string      `json:"warnings"`
+        } `json:"result"`
+}
+
+ // GetChainInfo returns information about the connected backend
+func (b *LuxRPC) GetChainInfo() (*bchain.ChainInfo, error) {
+        glog.V(1).Info("rpc: getblockchaininfo")
+
+         resCi := ResGetBlockChainInfo{}
+        err := b.Call(&CmdGetLuxBlockChainInfo{Method: "getblockchaininfo"}, &resCi)
+        if err != nil {
+                return nil, err
+        }
+        if resCi.Error != nil {
+                return nil, resCi.Error
+        }
+
+         glog.V(1).Info("rpc: getnetworkinfo")
+        resNi := ResGetNetworkInfo{}
+        err = b.Call(&CmdGetNetworkInfo{Method: "getnetworkinfo"}, &resNi)
+        if err != nil {
+                return nil, err
+        }
+        if resNi.Error != nil {
+                return nil, resNi.Error
+        }
+
+         // cant store difficulty here as there are two types
+        rv := &bchain.ChainInfo{
+                Bestblockhash: resCi.Result.Bestblockhash,
+                Blocks:        resCi.Result.Blocks,
+                Chain:         resCi.Result.Chain,
+                Headers:       resCi.Result.Headers,
+                SizeOnDisk:    resCi.Result.SizeOnDisk,
+                Subversion:    string(resNi.Result.Subversion),
+                Timeoffset:    resNi.Result.Timeoffset,
+        }
+        rv.Version = string(resNi.Result.Version)
+        rv.ProtocolVersion = string(resNi.Result.ProtocolVersion)
+        if len(resCi.Result.Warnings) > 0 {
+                rv.Warnings = resCi.Result.Warnings + " "
+        }
+        if resCi.Result.Warnings != resNi.Result.Warnings {
+                rv.Warnings += resNi.Result.Warnings
+        }
+        return rv, nil
+}
+
+
+ // NewLuxRPC returns new LuxRPC instance.
+func NewLuxRPC(config json.RawMessage, pushHandler func(bchain.NotificationType)) (bchain.BlockChain, error) {
+	b, err := btc.NewBitcoinRPC(config, pushHandler)
+	if err != nil {
+		return nil, err
+	}
+
+ 	s := &LuxRPC{
+		b.(*btc.BitcoinRPC),
+	}
+	s.RPCMarshaler = btc.JSONMarshalerV1{}
+	s.ChainConfig.SupportsEstimateSmartFee = true
+
+ 	return s, nil
+}
+
+ // Initialize initializes LuxRPC instance.
+func (b *LuxRPC) Initialize() error {
+	chainName, err := b.GetChainInfoAndInitializeMempool(b)
```

Comment:
The method `GetChainInfoAndInitializeMempool` was removed, please use here the following code instead:

```
	ci, err := b.GetChainInfo()
	if err != nil {
		return err
	}
	chainName := ci.Chain
```

Reply:
Thanks!